File: JavaZero/app/src/main/java/com/yagofellipe/javazero/herança.java
package com.yagofellipe.javazero;

class Scratch2 {
    public static void main(String[] args) {

        Cachorro cachorro = new Cachorro();
        cachorro.cor = "marrom";
        cachorro.dormir();

        class Cachorro {
            String cor;
            int tamanho;
            double peso;

            void correr(){
                System.out.println("correndo");
            }

            void dormir(){
                System.out.println("Dormindo");
            }
        }
    }
}
class Cachorro {
    String cor;
    int tamanho;
    double peso;

    void correr(){
        System.out.println("correndo");
    }

    void dormir(){
        System.out.println("Dormindo");
    }
}
class Passaro {
    String cor;
    int tamanho;
    double peso;

    void correr(){
        System.out.println("correndo");
    }

    void dormir(){
        System.out.println("Dormindo");
    }
}

File: kotlin/passageiro/src/main/java/com/yagofellipe/kotlin/Herança.kt
package com.yagofellipe.kotlin

abstract class Animal{  //classe abstrata
    var cor = "";
    var tamanho = "";
    var peso = 0.0;

    open fun correr(){
        print("correndo ");
    }
}

class Cao: Animal() { // class concreta
    override fun correr(){
        super.correr();
        println("como cachorro");
    }
}

fun main(){
    val cachorro = Cao();
    // val testabstrata = Animal(); não pode instanciar, pois é abstrata
    cachorro.correr();
}